### PR TITLE
Add functionality to suppress tokens during generation

### PR DESCRIPTION
**Pull request by abheesht17**
Resolves https://github.com/keras-team/keras-nlp/issues/975

### CODE REVIEW

**Comment by chenmoneygithub** [comment]
Thanks Abi! I am a little unsure if a blocklist of token ids could work as expected, mainly due to subword tokenizer. For common English curse words, they are usually one token so it could work, but it does not work for phrases or words would be split by subword tokenizer. Blocklisting words is a complex task as I see, we will need to discuss a bit more about it, thanks!

**Comment by mattdangerw** [comment]
Yeah, I think the short status update is we should leave this to discuss post 0.5 release. Logit manipulation generally is a good topic for discussion in our generation flow.

**Comment by abheesht17** [comment]
Gotcha! We will, however, have to do this for Whisper in any case: https://huggingface.co/openai/whisper-tiny/blob/main/generation_config.json#L126. I guess we can just modify the logits in the next() function for Whisper instead of changing the sampler API

**Comment by mattdangerw** [comment]
> Gotcha! We will, however, have to do this for Whisper in any case: https://huggingface.co/openai/whisper-tiny/blob/main/generation_config.json#L126. I guess we can just modify the logits in the next() function for Whisper instead of changing the sampler API

Good to know! I think whisper (like all the seq2seq stuff) we will leave out of 0.5, but does sounds like we need to think through this soon.

**Comment by sachinprasadhs** [comment]
@abheesht17 , Is this PR still relevant, if yes, could you please update the status of this PR. Thanks!
